tensorflow_lite_support/c/test/task/vision/image_classifier_test.cc

```diff
@@ -0,0 +1,302 @@
+                                    kMobileNetQuantizedWithMetadata)
+                               .data();
+
+  TfLiteImageClassifierOptions options = {0};
+  options.base_options.model_file.file_path = model_path;
+  options.base_options.compute_settings.num_threads = 3;
+  TfLiteImageClassifier* image_classifier =
+      TfLiteImageClassifierFromOptions(&options);
+  EXPECT_NE(image_classifier, nullptr);
+  TfLiteImageClassifierDelete(image_classifier);
+}
+
+TEST(CImageClassifierFromOptionsTest,
+     FailsWithClassNameBlackListAndClassNameWhiteList) {
+  const char* model_path = JoinPath("./" /*test src dir*/, kTestDataDirectory,
+                                    kMobileNetQuantizedWithMetadata)
+                               .data();
+
+  TfLiteImageClassifierOptions options = {0};
+  options.base_options.model_file.file_path = model_path;
+
+  char* class_name_blacklist[] = {"brambling"};
+  options.classification_options.class_name_blacklist.list =
+      class_name_blacklist;
+  options.classification_options.class_name_blacklist.length = 1;
+
+  char* class_name_whitelist[] = {"cheeseburger"};
+  options.classification_options.class_name_whitelist.list =
+      class_name_whitelist;
+  options.classification_options.class_name_whitelist.length = 1;
+
+  TfLiteImageClassifier* image_classifier =
+      TfLiteImageClassifierFromOptions(&options);
+  EXPECT_EQ(image_classifier, nullptr);
+  if (image_classifier) TfLiteImageClassifierDelete(image_classifier);
+}
+
+class CImageClassifierClassifyTest : public ::testing::Test {
+ protected:
+  void SetUp() override {
+    TfLiteImageClassifierOptions options = {0};
+    options.base_options.model_file.file_path =
+        JoinPath("./" /*test src dir*/, kTestDataDirectory,
+                 kMobileNetQuantizedWithMetadata)
+            .data();
+    image_classifier = TfLiteImageClassifierFromOptions(&options);
+
+    ASSERT_NE(image_classifier, nullptr);
+  }
+
+  void TearDown() override { TfLiteImageClassifierDelete(image_classifier); }
+  TfLiteImageClassifier* image_classifier;
+};
+
+TEST_F(CImageClassifierClassifyTest, SucceedsWithImageData) {
+  CImageData image_data = LoadImage("burger-224.png");
+
+  TfLiteFrameBuffer frame_buffer = {.dimension.width = image_data.width,
+                                    .dimension.height = image_data.height,
+                                    .buffer = image_data.pixel_data,
+                                    .format = kRGB};
+
+  TfLiteClassificationResult* classification_result =
+      TfLiteImageClassifierClassify(image_classifier, &frame_buffer);
+
+  CImageDataFree(&image_data);
+
+  ASSERT_NE(classification_result, nullptr) << "Classification Result is NULL";
+  EXPECT_TRUE(classification_result->size >= 1)
+      << "Classification Result size is 0";
+  EXPECT_NE(classification_result->classifications, nullptr)
+      << "Classification Result Classifications is NULL";
```

Comment:
Those messages look redundant. The `EXPECT_XXX` statements are already self-explained. You can remove these messages in this file.

Reply:
Removed messages